docs/embeddings/roboflow.md

```diff
@@ -0,0 +1,142 @@
+
+1. How to set up Roboflow Inference
+2. How to create a Chroma vector database
+3. How to calculate CLIP embeddings with Inference
+4. How to run a search query with Chroma
+
+Without further ado, let’s get started!
+
+## What is Roboflow Inference?
+
+[Roboflow Inference](https://inference.roboflow.com) is a scalable server through which you can run fine-tuned object detection, segmentation, and classification models, as well as popular foundation models such as CLIP.
+
+Inference handles all of the complexity associated with running vision models, from managing dependencies to maintaining your environment.
+
+Inference is trusted by enterprises around the world to manage vision models, with the hosted version powering millions of API calls each month.
+
+### Step #1: Set Up Roboflow Inference
+
+Inference runs in Docker and provides a HTTP interface through which to retrieve predictions.
+
+We will use Inference to calculate CLIP embeddings for our application.
+
+There are two ways to use Inference:
+
+1. On your device
+2. Through the Inference API hosted by Roboflow
+
+We will talk through both options, but we’ll start by showing how to install Inference on your own device. This is ideal if you want to deploy a model on your own infrastructure.
+
+To install inference, you will need Docker on your machine. Learn how to install Docker from the official Docker installation instructions. With Docker installed, you can install Inference with pip:
+
+```
+pip install inference-cli
+```
+
+Run the following command to start an Inference server:
+
+```
+inference server start
+```
+
+A message will appear that notes the server is starting. Inference will start running on `http://localhost:9001`.
+
+### Step #2: Create a Chroma Vector Database
+
+To load and save image embeddings into Chroma, we first need images to embed. In this guide, we are going to use the COCO 128 dataset, a collection of 128 images from the Microsoft COCO dataset. This dataset is available on Roboflow Universe, a community that has shared more than 250,000 public computer vision datasets.
+
+To download the dataset, visit the COCO 128 web page, click “Download Dataset” and save the dataset as a ZIP file following the on-screen instructions:
+
+![COCO 128 dataset](https://media.roboflow.com/coco128.png)
+
+We are going to use the images in the `train/images` folder of the dataset.
+
+Now that we have a dataset ready, we can create a vector database and start loading embeddings.
+
+Install the Chroma Python client with the following command:
+
+```
+pip install chromadb
+```
+
+Then, create a new Python file and add the following code:
+
+```python
+import chromadb
+import os
+from chromadb.utils.embedding_functions import RoboflowEmbeddingFunction
+import uuid
+
+client = chromadb.PersistentClient(path="database")
+
+collection = client.create_collection(name="images", metadata={"hnsw:space": "cosine"})
```

Comment:
rather than creating a collection, then separately calling the `RoboflowEmbeddingFunction`, you can just pass in the embedding function when you create the collection.

additionally, to save the user managing loading the images, we can use the `ImageDataLoader` - see the multi-modal docs here: https://docs.trychroma.com/multi-modal